frontend/src/utils/AuthService.ts

```diff
@@ -91,17 +96,30 @@ export default class AuthService {
     return this._authState;
   }
 
-  static createLoggedInEvent = (detail: LoggedInEventDetail): LoggedInEvent => {
-    return new CustomEvent(AuthService.loggedInEvent, { detail });
-  };
+  static createLoggedInEvent = (
+    detail?: LoggedInEvent["detail"]
+  ): LoggedInEvent =>
+    new CustomEvent(AuthService.loggedInEvent, {
+      bubbles: true,
+      composed: true,
+      detail,
+    });
 
-  static createNeedLoginEvent = (redirectUrl?: string): NeedLoginEvent => {
-    return new CustomEvent(AuthService.needLoginEvent, {
+  static createLogOutEvent = (detail?: LogOutEvent["detail"]): LogOutEvent =>
+    new CustomEvent(AuthService.logOutEvent, {
       bubbles: true,
       composed: true,
-      detail: { redirectUrl },
+      detail,
+    });
+
+  static createNeedLoginEvent = (
+    detail?: NeedLoginEvent["detail"]
+  ): NeedLoginEvent =>
+    new CustomEvent(AuthService.needLoginEvent, {
+      bubbles: true,
+      composed: true,
+      detail,
     });
```

Comment:
Similar here, if we supply `CustomEvent` with a type, it'll properly type check the `detail` param — It doesn't matter much here because there's an explicit return type and there's nothing else going on in the function, but it's also generally good practice to reduce `any`s wherever possible
```suggestion
  static createNeedLoginEvent = (
    detail?: NeedLoginEvent["detail"]
  ) =>
    new CustomEvent<NeedLoginEvent["detail"]>(AuthService.needLoginEvent, {
      bubbles: true,
      composed: true,
      detail,
    });
```